### ISSUE TITLE
Hola  [About jv-scraper]

### ISSUE THREAD

**Issue by cimcoz** [helkyar/Web-Scraper#1]
Hola Javier,

I am interested your below small project java part.
I import [jv-scraper](https://github.com/helkyar/Web-Scraper/tree/main/jv-scraper) porject in to eclipse and netbeans. I also guessed db structure. But project running and not adding data into db.
Is there trick for that? Or if you can share your correct database sql or structure I will be happy.

regards/adios.

**Comment by helkyar** [comment]
Funny how I didn't think about that. It should be everything ok now (assets folder). Also added the .zip export of the proyect, it may help.

**Comment by cimcoz** [comment]
Thanks for your quick reply. But I can't find assets folder right now. Are going to update later or already did it? Gracias..

**Comment by helkyar** [comment]
I thougth I did, but git is giving me some trouble. Done

**Comment by cimcoz** [comment]
It's fine now gracias. By the way only games table is updating if I change the league from link, teams table not updating or changing, bets table too. Are still developing this project or already completed? (Java part I mean)

**Comment by helkyar** [comment]
It's already functional for me, only needed some data about futball games and update it based on results. Bets and users tables are updated through other means. That said I'll work on this or similar proyects in the future, but don't know when.
I don't quite understand the problem about the tables update, is meant to change only when the web changes, if you could describe it more precicely I'll check it out.

**Comment by cimcoz** [comment]
Sure I can. Currently I updated database which provided sql.  When I run app onlu games table updating.

For example : I changed league link as :         final String URL = "https://www.marca.com/futbol/liga-turca/calendario.html"; from Spanish league to Turca (Turkish) League. Games table changed to Turkish teams but teams table stays same.

**Comment by helkyar** [comment]
Okey, following my code what you really want is store new teams. For that just uncoment the function "storeInDatabaseTeams(loc);" same goes for Games table. It was updating on Games (when it shouldn't) because the filter criteria is the loop index. Since it really doesn't make sense to destroy all previous data you should use an unique identifier (well maybe in your case it does but if not done thoroughly it will cause errors)

**Comment by helkyar** [comment]
Let me update it real quick

**Comment by helkyar** [comment]
There you go, code updated. Just uncomment the "store" methods and it should work (I didn't test it tho)

**Comment by cimcoz** [comment]
When I activated  storeInDatabaseGames(id, loc, vis, res); and storeInDatabaseTeams(loc); methods I got 
this errors related "game_id" as : java.sql.SQLException: Field 'game_id' doesn't have a default value. Which table has game_is column? games and/or teams?

**Comment by helkyar** [comment]
Since game_id is primary key in games it shouldn't be the source of the error. Try changing the order in witch the methods are called

**Comment by cimcoz** [comment]
Thanks. I got 
java.lang.NullPointerException
	at javascraper.Scraper.getCurrentGamesCount(Scraper.java:252)
	at javascraper.Scraper.main(Scraper.java:56) 
error right now. Regarding   id = getCurrentGamesCount(); method ..

**Comment by helkyar** [comment]
With the last update I made sure (at least I think I did) that id allways recives a number. If the issue persist make sure the last statement ("finally") returns a number.

**Comment by cimcoz** [comment]
You're right. I just deleted existing table so cannot find game_id. I inserted sample data and it's works. By the way I tried to make for different leagues, some leagues has 18 teams son I made different main methods and inserting into different tables as well. I wonder how can I use single method instead of 7 different leagues main methods (Esp, Ita, Ned, Tur, Ger, Eng and Fra) except DAO etc. Any idea for that?
I mean each league has different classes and one main method will run any league methods in Main.java.
